### ISSUE TITLE
Removing a FileVersionMeta entity also removes the FileVersion

### ISSUE THREAD

**Issue by Prokyonn** [sulu/sulu#7266]
| Q | A
| --- | ---
| Sulu Version | 2.5.11
| PHP Version | 8.2.14
| DB Version | MySQL 8.0.23

# Actual Behavior

Removing a `FileVersionMeta` (me_file_version_meta) entity from the database automatically removes the `FileVersion` (me_file_version) entity as well. This results in missing `FileVersions`.

# Expected Behavior

Removing `FileVersionMeta` only removes this entity and as long as there is still another `FileVersionMeta` (e.g. different locale) that references the `FileVersion`, the `FileVersion` should not be removed.

# Steps to Reproduce

Take a Sulu installation with multiple locales.
Save the file metadata in both languages, so that there are two `FileVersionMeta` entries for the same `FileVersion`.
Remove one of the `FileVersionMeta` entries and the `FileVersion` should still exist.

# Possible Solutions

Adjust the foreign key constraints

**Comment by alexander-schranz** [comment]
I think the issue is not the foreign key. As the foreign key only tells if `FileVersion` is removed the `FileVersionMeta` also should be removed and not an error is thrown when try to remove the `FileVersion`. The foreign key normally don't work into the other direction.

I think more the problem is on the doctrine level of casacing, so the `cascade-all` here: https://github.com/sulu/sulu/blob/51653362aa04f7762c4e45ff6b6306eb6cc7d89e/src/Sulu/Bundle/MediaBundle/Resources/config/doctrine/FileVersion.orm.xml#L38

We should just do `cascade-persist`, it looks like it only happens for the default meta?

**Comment by alexander-schranz** [comment]
@Prokyonn Ohhh you were right we have a foreign key in the other direction which is here: https://github.com/sulu/sulu/blob/51653362aa04f7762c4e45ff6b6306eb6cc7d89e/src/Sulu/Bundle/MediaBundle/Resources/config/doctrine/FileVersion.orm.xml#L41. We should remove it, so it hard fails if somebody removes the default meta without setting a new one 🙈 .

**Comment by alexander-schranz** [comment]
So we should do:

```diff
-<join-column name="idFileVersionsMetaDefault" referenced-column-name="id" nullable="true" on-delete="CASCADE" />
+<join-column name="idFileVersionsMetaDefault" referenced-column-name="id" nullable="true" /><!-- No cascade remove as a default meta is always required -->
```

because default meta is required always. As discussed with @Prokyonn there import need to take care of setting a new default meta. But as meta remove is not implemented in core the only change which is required remove the cascade here to trigger an error.

**Comment by redjanym** [comment]
> So we should do:
> 
> ```diff
> -<join-column name="idFileVersionsMetaDefault" referenced-column-name="id" nullable="true" on-delete="CASCADE" />
> +<join-column name="idFileVersionsMetaDefault" referenced-column-name="id" nullable="true" />
> ```
> 
> because default meta is required always. As discussed with @Prokyonn there import need to take care of setting a new default meta. But as meta remove is not implemented in core the only change which is required remove the cascade here to trigger an error.

If we remove the `on-delete="CASCADE"` the deletion of a specific FileVersionMeta will fail every time. By just changing:
```diff
-<cascade-all />
+<cascade-persist />
```

we can fix the problem.

**Comment by alexander-schranz** [comment]
@redjanym that is expected error we want to have.

**Comment by alexander-schranz** [comment]
This was fixed in #7282
